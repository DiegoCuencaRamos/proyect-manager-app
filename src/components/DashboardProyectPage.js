import React from 'react'
import { Link } from 'react-router-dom'
import { useSelector } from 'react-redux'

import Title from './Title'
import FiltersList from './FiltersList'
import ItemList from './ItemList'
import KanbanBoard from './KanbanBoard'

const DashboardProyectPage = () => {
    const visualizeMode = useSelector(state => state.filters.visualizeMode)
    let itemsLayout = <ItemList isProyect={true} />

    if (visualizeMode === 'list') {
        itemsLayout = <ItemList isProyect={true} />
    } else if (visualizeMode === 'kanban') {
        itemsLayout = <KanbanBoard isProyect={true}/>
    } else if (visualizeMode === 'schedule') {
        
    }

    return (
        <section>
            <Title 
                title={'Proyect Dashboard'} 
                description={'From here you will be able to visualice all your proyects'}
                isDashboard={true}
                isProyect={true}
            />
            <FiltersList />
            {itemsLayout}
        </section>
    )
}

export default DashboardProyectPage