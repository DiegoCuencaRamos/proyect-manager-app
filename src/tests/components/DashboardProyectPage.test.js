import React from 'react'
import { shallow } from 'enzyme'
import DashboardProyectPage from '../../components/DashboardProyectPage'

let wrapper

beforeEach(() => {
    wrapper = shallow(<DashboardProyectPage />)
})

test('Should render Dashboard page correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('Should set default layout state', () => {
    expect(wrapper.find('ItemsLayout').prop('layout')).toBe('')
})

test('Should set layout state', () => {
    const layoutData = 'list'
    wrapper.find('LayoutSelectors').prop('handleLayoutChange')(layoutData)
    expect(wrapper.find('ItemsLayout').prop('layout')).toBe(layoutData)
})