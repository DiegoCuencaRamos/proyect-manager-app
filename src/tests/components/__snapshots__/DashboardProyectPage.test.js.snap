// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Dashboard page correctly 1`] = `
<ContextProvider
  value={
    Object {
      "isProyect": true,
    }
  }
>
  <section>
    <Title
      description="From here you will be able to visualice all your proyects"
      isDashboard={true}
      title="Proyect Dashboard"
    />
    <LayoutSelectors
      handleLayoutChange={[Function]}
    />
    <ItemsLayout
      layout=""
    />
  </section>
</ContextProvider>
`;
